src/bin/signer-process-success.rs

```diff
@@ -0,0 +1,99 @@
+// Copyright 2024 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License,
+// Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+// or the MIT license (http://opensource.org/licenses/MIT),
+// at your option.
+// Unless required by applicable law or agreed to in writing,
+// this software is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR REPRESENTATIONS OF ANY KIND, either express or
+// implied. See the LICENSE-MIT and LICENSE-APACHE files for the
+// specific language governing permissions and limitations under
+// each license.
+
+use std::{
+    io,
+    io::{Read, Write},
+    path::PathBuf,
+};
+
+use clap::Parser;
+use openssl::{
+    hash::MessageDigest,
+    pkey::PKey,
+    rsa::{Rsa, RsaPrivateKeyBuilder},
+};
+
+#[derive(Parser, Debug)]
+#[clap(author, version, about)]
+struct ExternalSignerArgs {
+    #[clap(short, long, required = true)]
+    reserve_size: usize,
+    /// The algorithm used to sign the manifest.
+    #[clap(short, long, required = true)]
+    alg: String,
+    /// The path to the public key.
+    #[clap(short, long, required = true)]
+    sign_cert: PathBuf,
+}
+
+/// A simple program demonstrating how to write an example CLI tool that signs C2PA manifests.
+fn main() -> io::Result<()> {
+    let mut bytes_to_be_signed: Vec<u8> = vec![];
+    // 1. Read the bytes to be signed from this process' `stdin`.
+    io::stdin().read_to_end(&mut bytes_to_be_signed).unwrap();
+
+    // We provide a few arguments to this process via CLI args as [ExternalSignerArgs].
+    let _cli = ExternalSignerArgs::parse();
```

Comment:
Should we show or explain what these parameters are for or why we pass them?

Reply:
Resolved here: https://github.com/contentauth/c2patool/pull/169/commits/af21581b597ee4400e017530d5ec4a14258b2bdc